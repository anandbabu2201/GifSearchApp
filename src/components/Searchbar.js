import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { getGifs,getRandomGif,getTrendingGif,getDataChangeEvent } from "../actions/gifActions";

class Searchbar extends Component {

  state = {
    filterBy:'',
    searchBy:''
  }

  callGifData=(e)=>{
    this.setState({searchBy:e.target.value})
    this.props.getGifs(e.target.value)
    if(this.state.filterBy) this.setState({filterBy:''})
    this.props.getDataChangeEvent({
      filterBy:'',
      searchBy:e.target.value
    });
  }

  getTrendingGif= _=>{
    this.setState({filterBy:'Trending'})
    if(this.state.searchBy)
     this.setState({searchBy:''})
    this.props.getTrendingGif();
    this.props.getDataChangeEvent({
      filterBy:'Trending',
      searchBy:''
    });
  }

  getRandomGif = _ =>{
    this.setState({filterBy:'Random'})
    this.props.getRandomGif(this.state.searchBy);
    this.props.getDataChangeEvent({
      filterBy:'Random',
      searchBy:this.state.searchBy
    });
  }


  render() {
    return (
      <React.Fragment>
       <div className="component " >
         <div className="searchBox">
           <input name="gifSearch" className="inputSearch" type="text" placeholder="search GIF"
           onChange={this.callGifData} value={this.state.searchBy} />
          </div>
           <div className="actionButtons">
           <button className="button" onClick={this.getRandomGif} name="randomButton">Random Gif</button>
          <button className="button" onClick={this.getTrendingGif} name="trendingButton">Trending Gifs</button>
            </div>
          </div>
          {this.state.searchBy || this.state.filterBy ? <h2 className="heading">showing "{this.state.filterBy || this.state.searchBy}" Gif(s)</h2>:
            ''}
          </React.Fragment>
    );
  }
}

Searchbar.propTypes = {
  getGifs: PropTypes.func.isRequired,
  getTrendingGif: PropTypes.func.isRequired,
  getRandomGif: PropTypes.func.isRequired,
  getDataChangeEvent:PropTypes.func.isRequired
};


export default connect(null,
  { getGifs,getTrendingGif,getRandomGif,getDataChangeEvent }
)(Searchbar);
