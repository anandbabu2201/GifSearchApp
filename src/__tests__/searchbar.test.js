import React from 'react';
import {shallow} from 'enzyme';
import Searchbar from '../components/Searchbar';
import thunk from 'redux-thunk';
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore([ thunk ]);
let store ;
let wrapper;

beforeEach(() => {
  store = mockStore({});
  wrapper = shallow(<Searchbar store={store} />).shallow();
});

describe('Rendering searchbar component',() => {
  it('renders Searchbar component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('searchbar component rendered 3 div elements', () => {
    expect(wrapper.find('div')  ).toHaveLength(3);
  });
  it('searchbar component rendered input elements', () => {
    expect(wrapper.find('input')  ).toHaveLength(1);
  });
  it('searchbar component rendered button elements', () => {
    expect(wrapper.find('button')  ).toHaveLength(2);
  });
});

describe('sumulating actions on searchbar component',() => {

  it('it should call callGifData on change of text in input ', () => {
    const event = {
      target: { value: 'testvalue',scrollHeight:20 }
    };
    wrapper.find('input').simulate('change',event);
  });
  it('clicking on random gif', () => {
    wrapper.find('button').at(0).simulate('click');
  });
  it('clicking on Trending gif', () => {
    wrapper.find('button').at(1).simulate('click');
  });
  it('checking if the user entered any data and clear while click on TrendingGif ', () => {
    const event = {
      target: { value: 'testvalue' }
    };
    wrapper.find('input').simulate('change',event);
    wrapper.update();
    wrapper.find('button').at(1).simulate('click');
  });
  it('checking if the user filtered gifs by trending data and clear while typing', ()=> {
    const event = {
      target: { value: 'testvalue' }
    };
    wrapper.find('button').at(1).simulate('click');
    wrapper.update();
    wrapper.find('input').simulate('change',event);
  });

});
