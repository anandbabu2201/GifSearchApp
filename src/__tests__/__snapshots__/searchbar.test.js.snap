// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering searchbar component renders Searchbar component 1`] = `ShallowWrapper {}`;
